src/main/test/CocktailControllerTest.java

```diff
@@ -0,0 +1,138 @@
+
+    private List<Cocktail> cocktailList = Arrays.asList(cocktail);
+
+    private final static String COCKTAIL_NAME = "Sex on the Beach Cocktail Recipe";
+    private final static String COCKTAIL_AUTHOR_NAME = "Iurie Railean";
+    private final static String INGRDIENT_NAME = "Wodka";
+
+    private final static String COCKTAIL_WEB_ARRAY_JSON = "[{\"id\":1,\"title\":\"Sex on the Beach Cocktail Recipe\",\"announce\":\"So I have to say one of the best drink's I've drunk so far has become my favorite drink as well\",\"publicationDate\":null,\"prepTime\":1,\"cookingTime\":1,\"author\":{\"id\":1,\"name\":\"Iurie Railean\"},\"image\":{\"id\":1,\"path\":\"http//google.com\",\"title\":\"\",\"alt\":\"\"},\"ingredients\":[{\"id\":1,\"title\":\"Wodka\",\"description\":\"4 cl Wodka\"},{\"id\":2,\"title\":\"Pfirsich Likör\",\"description\":\"4 cl Pfirsich Likör\"},{\"id\":3,\"title\":\"cranberry juice or Cranberry Rectal\",\"description\":\"8 cl cranberry juice or Cranberry Rectal\"},{\"id\":4,\"title\":\"orange juice\",\"description\":\"8 cl of orange juice\"},{\"id\":5,\"title\":\"ice\",\"description\":\"ice cubes\"}]}]";
+
+    private final static String COCKTAIL_SEX_ARRAY_JSON = "[{\"id\":1,\"title\":\"Sex on the Beach Cocktail Recipe\",\"announce\":\"So I have to say one of the best drink's I've drunk so far has become my favorite drink as well\",\"publicationDate\":null,\"author\":{\"name\":\"Iurie Railean\"},\"ingredients\":[{\"title\":\"Wodka\",\"description\":\"4 cl Wodka\"},{\"title\":\"Pfirsich Likör\",\"description\":\"4 cl Pfirsich Likör\"},{\"title\":\"cranberry juice or Cranberry Rectal\",\"description\":\"8 cl cranberry juice or Cranberry Rectal\"},{\"title\":\"orange juice\",\"description\":\"8 cl of orange juice\"},{\"title\":\"ice\",\"description\":\"ice cubes\"}],\"prepTimeMinute\":0,\"cookingTime\":1,\"image\":{\"path\":\"http//google.com\",\"title\":\"\",\"alt\":\"\"}}]";
+    private final static String COCKTAIL_SEX_JSON = "{\"id\":1,\"title\":\"Sex on the Beach Cocktail Recipe\",\"announce\":\"So I have to say one of the best drink's I've drunk so far has become my favorite drink as well\",\"publicationDate\":null,\"author\":{\"name\":\"Iurie Railean\"},\"ingredients\":[{\"title\":\"Wodka\",\"description\":\"4 cl Wodka\"},{\"title\":\"Pfirsich Likör\",\"description\":\"4 cl Pfirsich Likör\"},{\"title\":\"cranberry juice or Cranberry Rectal\",\"description\":\"8 cl cranberry juice or Cranberry Rectal\"},{\"title\":\"orange juice\",\"description\":\"8 cl of orange juice\"},{\"title\":\"ice\",\"description\":\"ice cubes\"}],\"prepTimeMinute\":0,\"cookingTime\":1,\"image\":{\"path\":\"http//google.com\",\"title\":\"\",\"alt\":\"\"}}";
+
+    @Test
+    public void findCocktailsByIngredientsTitleTest() throws Exception {
+        when(cocktailService.getCocktailByIngredient(INGRDIENT_NAME))
+                .thenReturn(cocktailList);
+
+        mockMvc.perform(get("/cocktails/ingredients/{name_ingredient}", INGRDIENT_NAME)
```

Comment:
In unit tests it is very important to be explicit in what your mock parameters and what is the expected outcome. Therefore if you test an endpoint by name, specify the name explicitly, not as a call to getter of your fake cocktail object.